### PR TITLE
Data from widgets use new datalocator based editor

### PR DESCRIPTION
**Pull request by apepper**
Needs content: https://edit-beta.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://scrivito-attributename.scrivito-portal-app.pages.dev/de/mein-tynacoon-bd53fa4ef38c1c40?_scrivito_workspace_id=kc8f9712c22a4960&_scrivito_display_mode=diff

Also: After this PR is deployed, please also merge https://edit.scrivito.com/6d226c03f32a8ea4e8ae8e5cbe6c6e2c~https://pisa.scrivito-portal-app.pages.dev/6d226c03f32a8ea4e8ae8e5cbe6c6e2c/en?_scrivito_workspace_id=j493b39134af943b&_scrivito_display_mode=view

### CODE REVIEW

File: src/Widgets/DataFormNumberWidget/DataFormNumberWidgetComponent.tsx [hunk 1]
```diff
@@ -9,17 +9,17 @@ import { OverlayTrigger, Popover } from 'react-bootstrap'
 import { DataFormNumberWidget } from './DataFormNumberWidgetClass'
 
 provideComponent(DataFormNumberWidget, ({ widget }) => {
-  const dataItem = useData().dataItem()
+  const dataItemAttribute = useData().dataItemAttribute()
```
Comment (dcsaszar) [inline src/Widgets/DataFormNumberWidget/DataFormNumberWidgetComponent.tsx:12]: * [ ] FYI many `const dataItemAttribute` are only single-use temporary variables and could be inlined.

File: src/Widgets/DataFormNumberWidget/DataFormNumberWidgetComponent.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ provideComponent(DataFormNumberWidget, ({ widget }) => {
         className="form-control"
         defaultValue={defaultValue}
         id={id}
-        name={attributeName}
+        name={attributeName ?? ''}
```
Comment (dcsaszar) [inline src/Widgets/DataFormNumberWidget/DataFormNumberWidgetComponent.tsx:66]: * [ ] FYI maybe `{attributeName ?? undefined}` expresses the state more clearly. Not sure how exactly form submissions behave with nameless vs. empty-named fields.

File: src/Widgets/index.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { initNeoletterFormWidgets } from 'scrivito-neoletter-form-widgets'
 import.meta.glob(['./**/*WidgetClass.ts', './**/*WidgetComponent.tsx'], {
   eager: true,
 })
+import './migrateAttributeNameToData'
```
Comment (dcsaszar) [inline src/Widgets/index.ts]: * [ ] FYI Not reviewing. If there are any problems with the migration they can be fixed without touching code. I'd prefer to not have this in the history.

**Review by dcsaszar — APPROVED** [review]
LGTM. Some comments.

File: src/Widgets/DataFormNumberWidget/DataFormNumberWidgetComponent.tsx (hunk 2 above)
Comment (apepper) [reply src/Widgets/DataFormNumberWidget/DataFormNumberWidgetComponent.tsx:66]: I think both will fail due to

https://github.com/Scrivito/scrivito-portal-app/blob/b5b853dde539cd5a87e857034d4ea51c7c703bc0/src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx#L113-L114

**Comment by apepper** [comment]
REVIEW ([76507c2](https://github.com/Scrivito/scrivito-portal-app/pull/610/commits/76507c2374ae9ad77f22a6d629aab3973ed3c8d5))

**Review by dcsaszar — APPROVED** [review]
LGTM

**Pull request merged**